File: src/Tutor.WebClient/src/features/main/types/student.ts
import { uuid } from "../lib"
import { Lesson } from "./lesson"

export type Student = {
  id: string
  name: string
  defaultPrice: number
  avatarUrl: string
  description: string
  lessons: Lesson[]
}

export function createStudent(student?: Partial<Student>): Student {
  return { id: uuid(), name: "", avatarUrl: "", description: "", lessons: [], defaultPrice: 1500, ...student }
}

File: src/Chroniq.Models/Student.cs
namespace Chroniq.Models;

public class Student
{
    public required Guid Id { get; set; }
    public required string Name { get; set; }
    public required string? Description { get; set; }
    public required string? AvatarUrl { get; set; }
    public required int DefaultPrice { get; set; }
    public required bool IsArchived { get; set; }
    public required List<Lesson> Lessons { get; set; } = [];
    public required User User { get; set; }
}

public class StudentSiteDto
{
    public required Guid Id { get; set; }
    public required string Name { get; set; }
    public required string? Description { get; set; }
    public required string? AvatarUrl { get; set; }
    public required int DefaultPrice { get; set; }
    public required bool IsArchived { get; set; }
}

public static class StudentExtensions
{
    public static StudentSiteDto ToSiteDto(this Student student)
    {
        return new StudentSiteDto
        {
            Id = student.Id,
            Name = student.Name,
            Description = student.Description,
            AvatarUrl = student.AvatarUrl,
            DefaultPrice = student.DefaultPrice,
            IsArchived = student.IsArchived
        };
    }
}